openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityExportModalProvider/EntityExportModalProvider.component.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { Form, Input, Modal } from 'antd';
+import { AxiosError } from 'axios';
+import React, { ReactNode, useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { getCurrentLocaleDate } from 'utils/TimeUtils';
+import { showErrorToast } from 'utils/ToastUtils';
+import {
+  EntityExportModalContextProps,
+  ExportData,
+} from './EntityExportModalProvider.interface';
+
+const EntityExportModalContext =
+  React.createContext<EntityExportModalContextProps>(
+    {} as EntityExportModalContextProps
+  );
+
+export const EntityExportModalProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const [form] = Form.useForm();
+  const { t } = useTranslation();
+  const [exportData, setExportData] = useState<ExportData | null>(null);
+
+  const handleCancel = () => {
+    setExportData(null);
+  };
+
+  const showModal = (data: ExportData) => {
+    setExportData(data);
+  };
+
+  /**
+   * Creates a downloadable file from csv string and download it on users system
+   * @param data - csv string
+   */
+  const handleDownload = (data: string, fileName: string) => {
+    const element = document.createElement('a');
+
+    const file = new Blob([data], { type: 'text/plain' });
+
+    element.textContent = 'download-file';
+    element.href = URL.createObjectURL(file);
+    element.download = `${fileName}.csv`;
+    document.body.appendChild(element);
+    element.click();
+
+    URL.revokeObjectURL(element.href);
+    document.body.removeChild(element);
+  };
+
+  const handleExport = async ({ fileName }: { fileName: string }) => {
+    if (exportData === null) {
+      Promise.reject(t('message.api-must-provide-for-export'));
+
+      return;
+    }
```

Comment:
We have check for exportData before showing modal so don't think we have a case where the function is being called with exportData being null.
We can have then condition but no need to throw any error i belive